modules/tile-converter/src/i3s-converter/helpers/geometry-converter.ts

```diff
@@ -526,38 +530,77 @@ function convertMesh(
         cartographicOrigin,
         cartesianModelMatrix,
         nodeMatrix: matrix,
-        indices: primitive.indices?.value,
+        indices,
         attributeSpecificTransformation: transformVertexNormals,
         useCartesianPositions: false
       })
     );
     outputAttributes.texCoords = concatenateTypedArrays(
       outputAttributes.texCoords,
-      flattenTexCoords(
-        attributes.TEXCOORD_0 && attributes.TEXCOORD_0.value,
-        primitive.indices?.value
-      )
+      flattenTexCoords(attributes.TEXCOORD_0 && attributes.TEXCOORD_0.value, indices)
     );
 
     outputAttributes.colors = concatenateTypedArrays(
       outputAttributes.colors,
-      flattenColors(attributes.COLOR_0, primitive.indices?.value)
+      flattenColors(attributes.COLOR_0, indices)
     );
 
     if (materialUvRegion) {
       outputAttributes.uvRegions = concatenateTypedArrays(
         outputAttributes.uvRegions,
-        createUvRegion(materialUvRegion, primitive.indices?.value)
+        createUvRegion(materialUvRegion, indices)
       );
     }
 
     outputAttributes.featureIndicesGroups = outputAttributes.featureIndicesGroups || [];
     outputAttributes.featureIndicesGroups.push(
-      flattenBatchIds(getBatchIds(attributes, primitive, images), primitive.indices?.value)
+      flattenBatchIds(getBatchIds(attributes, primitive, images), indices)
     );
   }
 }
 
+type TypedArrayConstructor =
+  | Uint8ArrayConstructor
+  | Int8ArrayConstructor
+  | Uint16ArrayConstructor
+  | Int16ArrayConstructor
+  | Int32ArrayConstructor
+  | Uint32ArrayConstructor
+  | Float32ArrayConstructor
+  | Float64ArrayConstructor;
+
+/**
+ * Converts TRIANGLE-STRIPS to independent TRIANGLES
+ * @param {MeshPrimitive} primitive - the primitive to get the indices from
+ * @returns indices of vertices of the independent triangles
+ */
+function getIndices(primitive: MeshPrimitive): TypedArray {
```

Comment:
Not necessary, but there is a module loaders.gl/math that has some similar logic. Might put this there, or use something from there.